# 데이터분석방법론2 - 제6강 과제

### 학    번 : 201685-010100
###  성    명 : 지 용 기
###  연 락 처 : 010-9828-0332, braveji@hanmail.net
---

## 문제 5.1> 참게 자료에 대해(http://www.stat.ufl.edu/~aa/intro-cda/appendix.html), 무게와 너비를 예측변수로 하는 모형을 적합하라.

In [5]:
library(dplyr)


Attaching package: 'dplyr'

The following object is masked from 'package:stats':

    filter

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union



In [7]:
crab <- read.csv('chap06_report_01.csv', head=T)
head(crab)
summary(crab)

color spine width satell weight
1     3     3  28.3      8   3050
2     4     3  22.5      0   1550
3     2     1  26.0      9   2300
4     4     3  24.8      0   2100
5     4     3  26.0      4   2600
6     3     3  23.8      0   2100

     color           spine           width          satell           weight    
 Min.   :2.000   Min.   :1.000   Min.   :21.0   Min.   : 0.000   Min.   :1200  
 1st Qu.:3.000   1st Qu.:2.000   1st Qu.:24.9   1st Qu.: 0.000   1st Qu.:2000  
 Median :3.000   Median :3.000   Median :26.1   Median : 2.000   Median :2350  
 Mean   :3.439   Mean   :2.486   Mean   :26.3   Mean   : 2.919   Mean   :2437  
 3rd Qu.:4.000   3rd Qu.:3.000   3rd Qu.:27.7   3rd Qu.: 5.000   3rd Qu.:2850  
 Max.   :5.000   Max.   :3.000   Max.   :33.5   Max.   :15.000   Max.   :5200  

In [12]:
crab_add <- mutate( crab , has_satell =  ifelse( satell > 0 , 1 , 0)  )
crab_add.glm <- glm( has_satell ~ color + spine + width + weight  , binomial(link="logit"),   data=crab_add )
summary( crab_add.glm  )


Call:
glm(formula = has_satell ~ color + spine + width + weight, family = binomial(link = "logit"), 
    data = crab_add)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.1928  -0.9465   0.4919   0.8682   1.9892  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)  
(Intercept) -7.0078436  3.8034845  -1.842   0.0654 .
color       -0.5915286  0.2417241  -2.447   0.0144 *
spine        0.2716837  0.2410407   1.127   0.2597  
width        0.2732525  0.1893304   1.443   0.1489  
weight       0.0007949  0.0006917   1.149   0.2505  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 225.76  on 172  degrees of freedom
Residual deviance: 186.66  on 168  degrees of freedom
AIC: 196.66

Number of Fisher Scoring iterations: 4


### a. 예측식을 구하고 효과를 해석하라. 

- logit[ P( Y = 1) ] = -7.0078436 + -0.5915286 core +  0.2716837 spine  + 0.2732525  width  + 0.0007949 weight  으로 예측식이 나왔지만, Wald 검정방법으로는 유의한 변수는 color 뿐임.
- 그래서, logit[ P( Y = 1) ] = -7.0078436 + -0.5915286 core 임.
- color가 한 단위 증가할수록  e ^ ( -0.5915286  ) =  0.553 만큼 오즈비가 내려감.

In [16]:
exp( -0.5915286  ) 

[1] 0.5534806

### b. 어떤 예측변수도 반응변수에 영향을 주지 않는다는 가설에 대한 가능도비 검정을 실시하고 결과를 해석하라. 

In [18]:
crab_add.glm.anova <- anova( crab_add.glm,  test="Chisq")
crab_add.glm.anova

Analysis of Deviance Table

Model: binomial, link: logit

Response: has_satell

Terms added sequentially (first to last)


       Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    
NULL                     172     225.76              
color   1  12.4607       171     213.30 0.0004156 ***
spine   1   1.0460       170     212.25 0.3064362    
width   1  24.2360       169     188.02 8.522e-07 ***
weight  1   1.3552       168     186.66 0.2443640    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

- 가능도비 검정에서는 color와 width 변수가 유의미하고 spine과  weight 는 유의미하지 않음.

### c. 각 변수의 부분효과에 대해 각각 가능도비 검정을 실시하라. 왜 (b)의 검정방법에서는 매우 유의하게 나오지만 각각의 검정에서는 그렇지 않은가?

In [19]:
crab_add.glm_color  <- glm( has_satell ~ color , binomial(link="logit"),   data=crab_add )
crab_add.glm_color.anova <- anova( crab_add.glm_color,  test="Chisq")
crab_add.glm_color.anova

Analysis of Deviance Table

Model: binomial, link: logit

Response: has_satell

Terms added sequentially (first to last)


      Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    
NULL                    172     225.76              
color  1   12.461       171     213.30 0.0004156 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

In [20]:
crab_add.glm_spine  <- glm( has_satell ~ spine , binomial(link="logit"),   data=crab_add )
crab_add.glm_spine.anova <- anova( crab_add.glm_spine,  test="Chisq")
crab_add.glm_spine.anova

Analysis of Deviance Table

Model: binomial, link: logit

Response: has_satell

Terms added sequentially (first to last)


      Df Deviance Resid. Df Resid. Dev Pr(>Chi)
NULL                    172     225.76         
spine  1  0.13423       171     225.62   0.7141

In [21]:
crab_add.glm_width  <- glm( has_satell ~ width , binomial(link="logit"),   data=crab_add )
crab_add.glm_width.anova <- anova( crab_add.glm_width,  test="Chisq")
crab_add.glm_width.anova

Analysis of Deviance Table

Model: binomial, link: logit

Response: has_satell

Terms added sequentially (first to last)


      Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    
NULL                    172     225.76              
width  1   31.306       171     194.45 2.204e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

In [22]:
crab_add.glm_weight  <- glm( has_satell ~ weight , binomial(link="logit"),   data=crab_add )
crab_add.glm_weight.anova <- anova( crab_add.glm_weight,  test="Chisq")
crab_add.glm_weight.anova

Analysis of Deviance Table

Model: binomial, link: logit

Response: has_satell

Terms added sequentially (first to last)


       Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    
NULL                     172     225.76              
weight  1   30.021       171     195.74 4.273e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

-  spine 데이터만 빼고는 모두 유의미하게 나옴.
- 전체로 했을때와 부분변수로만 했을때와 차이가 있는 것은 변수들간의 교호작용 효과와 변수들간의 다중공분산성이 있기 때문임.

## 문제 5.3> 참게 자료에서 너비(W), 색깔(C)과 등뼈상태(S)를 예측변수로 사용한 가장 복잡한 모형을 가지고 후진제거법을 적용해보자. C*S*W는 세 예측변수 각각에 대한 주효과와 세 개 의 이차 교호작용과 하나의 삼차 교호작용을 포함하는 모형이다. <표 5.9>는 이 모형과 여러 가지의 간단한 모형들에 대한 적합 결과를 보여준다.

### a. 이 모형과 삼차 교호작용만을 제거한 모형을 비교하는 가능도비 검정을 시행하라. 삼차 교호작용항은 이 모형에서 제거될 수 있는가?

### b. 다음 단계에서 만약 하나의 항을 더 제거해야 한다면 모형 C*S+C*W을 선택해야 되는 이유를 설명하라

### c. 이 단계에서 이 모형과 S+C*W을 비교한 결과 이탈도는 8.0 (df=6 (P-값=0.24))이다. W+C*S와 비교한 결과 이탈도는 3.9 (df=3 (P-값=0.27))이다. 어떤 항을 제거해야 하는가?

### d. 마지막으로 앞에서 선택한 모형을 주효과 모형 C+S+W와 비교하라. 주효과 모형으로 축소시킬 수 있는가?

### e. 표에 있는 모형들 중에서 어떤 모형이 AIC의 기준에서 가장 적절한가?